File: shared/src/main/java/xyz/klinker/messenger/shared/service/SimpleLifetimeSubscriptionCheckService.java
package xyz.klinker.messenger.shared.service;

import xyz.klinker.messenger.shared.util.billing.ProductPurchased;

public class SimpleLifetimeSubscriptionCheckService extends SimpleSubscriptionCheckService {
    @Override
    protected void handleBestProduct(ProductPurchased best) {
        if (best.getProductId().equals("lifetime")) {
            writeLifetimeSubscriber();
        }
    }
}

File: shared/src/main/java/xyz/klinker/messenger/shared/service/SimpleLifetimeSubscriptionCheckService.kt
package xyz.klinker.messenger.shared.service

import xyz.klinker.messenger.shared.util.billing.ProductPurchased

class SimpleLifetimeSubscriptionCheckService : SimpleSubscriptionCheckService() {
    override fun handleBestProduct(best: ProductPurchased) {
        if (best.productId == "lifetime") {
            writeLifetimeSubscriber()
        }
    }
}
